### PR TITLE
bazel: reference 1.31

### PR DESCRIPTION
**Pull request by nfuden**
bump to 1.31

### CODE REVIEW

**Comment by EItanya** [comment]
My change isn't in `1.31` :(, it just merged today

**Comment by nfuden** [comment]
Converting to draft to pull up a new branch so as to not blow away commits on merge

File: bazel/extensions/extensions_build_config.bzl [hunk 1]
```diff
@@ -88,6 +87,7 @@ EXTENSIONS = {
     #
 
     "envoy.matching.inputs.application_protocol":       "//source/extensions/matching/network/application_protocol:config",
+    # Ideally these would be split up. We'll do so if anyone cares.
```
Comment (ashishb-solo) [inline bazel/extensions/extensions_build_config.bzl:90]: this looks fine to me, but let's chat about it. to be completely honest, i'm not 100% sure exactly how to review the changes in this file.

File: bazel/repositories.bzl [hunk 2]
```diff
@@ -67,8 +67,7 @@ def _repository_impl(name, **kwargs):
 
 def envoy_gloo_dependencies():
     _repository_impl("envoy", patches=[
-        "@envoy_gloo//bazel/foreign_cc:0001-otel-exporter-status-code-fix.patch",
-        "@envoy_gloo//bazel/foreign_cc:0002-ratelimit-filter-state-hits-addend.patch",
+        "@envoy_gloo//bazel/foreign_cc:0001-ratelimit-filter-state-hits-addend.patch",
```
Comment (ashishb-solo) [inline bazel/repositories.bzl:70]: what's the plan for the hits-addend patch? `0001-ratelimit-filter-state-hits-addend.patch` is currently an empty file

**Review by ashishb-solo — COMMENTED** [review]
a couple questions. i think we should be able to merge this without too many additional changes

File: bazel/extensions/extensions_build_config.bzl (hunk 1 above)
Comment (nfuden) [reply bazel/extensions/extensions_build_config.bzl:90]: Process was I copied over things and then commented out new things that we dnot think we are going to use

File: bazel/repositories.bzl (hunk 2 above)
Comment (nfuden) [reply bazel/repositories.bzl:70]: Ugh thanks. Yeah thats what reviews are for. Updating

File: bazel/extensions/extensions_build_config.bzl (hunk 1 above)
Comment (ashishb-solo) [reply bazel/extensions/extensions_build_config.bzl:90]: sounds good. i read [this section](https://github.com/envoyproxy/envoy/blob/main/bazel/README.md#enabling-and-disabling-extensions) over to better understand what we're doing here and i think all the changes seem reasonable. thanks!

**Review by ashishb-solo — APPROVED** [review]

**Pull request merged**